seeds: Add additional seed source and bump uptime requirements for Onion and I2P nodes

virtu opened the pull request:
This builds on #30008 and adds data [exported](https://github.com/virtu/seed-exporter) by [my crawler](https://github.com/virtu/p2p-crawler) an additional source for seed nodes. Data covers all supported network types.

[edit: Added Luke's seeder as input as well.]

### Motivation
- Further decentralizes the seed node selection process (in the long term potentially enabling an _n_-source threshold for nodes to prevent a single source from entering malicious nodes)
- No longer need to manually curate seed node list for any network type: See last paragraph of OP in #30008. My crawler has been [discovering the handful of available cjdns nodes](https://21.ninja/reachable-nodes/nodes-by-net-type/) for around two months, all but one of which meet the reliability criteria.
- Alignment of uptime requirements for Onion and I2P nodes with those of clearnet nodes to 50%: If I'm reading the code correctly, seeders appear to optimize for up-to-dateness by using [lower connection timeouts](https://github.com/achow101/dnsseedrs/blob/3c1a63c6723819871d76fe0fbd2155fe5a5bb171/src/crawl.rs#L349) than [Bitcoin Core](https://github.com/bitcoin/bitcoin/blob/bc87ad98543299e1990ee1994d0653df3ac70093/src/netbase.cpp#L40C27-L40C48) to maximize throughput. Since my crawler does not have the same timeliness requirements, it opts for accuracy by using generous timeouts. As a result, its data contains additional eligible Onion (and other darknet nodes), as is shown in the histogram below. Around 4500 Onion nodes are discovered so far (blue); my data adds ~6400 more (orange); ~ 1500 nodes take longer than the default 20-second Bitcoin Core timeout and won't qualify as "good".

![Connection time histogram for Onion nodes](https://github.com/user-attachments/assets/c3513604-aa48-4c75-b51d-13421eaed9eb)

Here's the current results with 512 nodes for all networks except cjdns:
<details>
<summary>Using the extra data</summary>

```
IPv4   IPv6  Onion  I2P    CJDNS Pass
10335   2531  11545   1589     10 Initial
10335   2531  11545   1589     10 Skip entries with invalid address
5639   1431  11163   1589      8 After removing duplicates
5606   1417  11163   1589      8 Enforce minimal number of blocks
5606   1417  11163   1589      8 Require service bit 1
4873   1228  11163   1589      8 Require minimum uptime
4846   1225  11161   1588      8 Require a known and recent user agent
4846   1225  11161   1588      8 Filter out hosts with multiple bitcoin ports
512    512    512    512      8 Look up ASNs and limit results per ASN and per net
```
</details>
<details>
<summary>Before</summary>

```
IPv4   IPv6  Onion  I2P    CJDNS Pass
5772   1323    443      0      2 Initial
5772   1323    443      0      2 Skip entries with invalid address
4758   1110    443      0      2 After removing duplicates
4723   1094    443      0      2 Enforce minimal number of blocks
4723   1094    443      0      2 Require service bit 1
3732    867    443      0      2 Require minimum uptime
3718    864    443      0      2 Require a known and recent user agent
3718    864    443      0      2 Filter out hosts with multiple bitcoin ports
 512    409    443      0      2 Look up ASNs and limit results per ASN and per net
```
</details>

### To dos
- [x] Remove manual nodes and update README
- [x] Mark nodes with connection times exceeding Bitcoin Core's default as bad in [exporter](https://github.com/virtu/seed-exporter): [done](https://github.com/virtu/seed-exporter/pull/12)
- [x] Regenerate mainnet seeds
- [x] Rebase, then remove WIP label once #30008 gets merged

virtu committed:
seeds: Pull nodes from virtu's crawler

Pull additional nodes from virtu's crawler. Data includes sufficient
Onion and I2P nodes to align the uptime requirements for these networks
to that of clearnet nodes (i.e., 50%). Data also includes more than
three times the number of CJDNS nodes currently hardcoded into
nodes_main_manual.txt, so hardcoded nodes becomes obsolete.

achow101 commented on contrib/seeds/README.md:
```diff
@@ -8,13 +8,15 @@ and remove old versions as necessary (at a minimum when SeedsServiceFlags()
 curl https://bitcoin.sipa.be/seeds.txt.gz | gzip -dc > seeds_main.txt
 curl https://mainnet.achownodes.xyz/seeds.txt.gz | gzip -dc >> seeds_main.txt
 curl https://testnet.achownodes.xyz/seeds.txt.gz | gzip -dc > seeds_test.txt
+curl https://21.ninja/seeds.txt.gz | gzip -dc > seeds_main.txt
```
This will overwrite instead of append.

virtu commented on contrib/seeds/nodes_main.txt:
```diff
@@ -630,1118 +747,166 @@
-[2a12:a302:1:a180::b5ca]:8333 # AS23959
-[2c0f:f4a8:b:b108:807d:b2d6:9146:38be]:8333 # AS37254
-[2c0f:f4a8:b:b108:c458:5c61:dcca:cb10]:8333 # AS37254
-iy7go4454pb4p2zmnkwrgsi6v6oqv53zxnmalz6rnfjemxftapfa.b32.i2p:0
```
I hadn't regenerated to node lists because I wanted to wait until Ava's seeder resumed exporting good I2P node data. I have now pushed a preliminary export anyway. Everything should be there.

luke-jr commented on contrib/seeds/README.md:
```diff
@@ -8,13 +8,19 @@ and remove old versions as necessary (at a minimum when SeedsServiceFlags()
+curl https://mainnet.achownodes.xyz/seeds.txt.gz | gzip -dc >> seeds_main.txt
+curl https://testnet.achownodes.xyz/seeds.txt.gz | gzip -dc > seeds_test.txt
+curl https://21.ninja/seeds.txt.gz | gzip -dc >> seeds_main.txt
+curl https://raw.githubusercontent.com/fjahr/asmap-data/main/latest_asmap.dat > asmap-filled.dat
```
Mine is here if you want to add it:

https://luke.dashjr.org/programs/bitcoin/files/charts/seeds.txt

virtu replied:
nice, added!

virtu committed:
seeds: Pull nodes from Luke's seeder

Pull additional nodes from Luke's seeder to further decentralize the
generation of seed nodes.

virtu committed:
seeds: Regenerate mainnet seeds

Regenerate mainnet seeds from new sources without the need for hardcoded
data. Result has 512 nodes from each network type except cjdns, for
which only eight nodes were found that match the seed node criteria.

virtu commented:
> I was manually curating I2P nodes based on trusted colleagues (akin to addnode peer selection), filtered by connection reliability and regularly seeing blocks from them. (Edit: I now see that you have ones that I'd recommend, so seems good. There were a couple that were missing, but I see that they, like me, began pruning.)

Good to know the automatic process is getting all of cjdns nodes you were tracking (modulo pruning).

Data from @luke-jr's seed is now included as well. As before: 512 nodes for each network type except cjdns, for which there are eight matching the seed node criteria.

achow101 commented:
ACK b061b3510585a1fe113cc9d1af65852b155aba45

Instruction updates look correct; did not check any of the changed seeds but did check that i2p and onion mainnet seeds have been readded.

If anyone has a testnet crawler, we probably want to have additional sources for testnet and add more i2p and onion testnet seeds.

fjahr reviewed (commented):
utACK b061b3510585a1fe113cc9d1af65852b155aba45

I have reviewed the changes and verified that the included files generate the included `chainparamsseeds.h`. I have not tested the seed nodes but I did a quick plausibility check on the txt file changes.

Fun side note: TIL that there is a stable node with a single digit ASN, AS9, which is Carnegie Mellon University.
